ibis/backends/flink/registry.py

```diff
@@ -207,6 +267,58 @@ def _floor_divide(translator: ExprTranslator, op: ops.Node) -> str:
     return f"FLOOR(({left}) / ({right}))"
 
 
+def _array_index(translator: ExprTranslator, op: ops.arrays.ArrayIndex):
+    table_column = op.arg
+    index = op.index
+
+    table_column_translated = translator.translate(table_column)
+    index_translated = translator.translate(index)
+
+    return f"{table_column_translated} '[' {index_translated} ']'"
+
+
+def _day_of_week_index(translator: ExprTranslator, op: ops.Node) -> str:
+    arg = translator.translate(op.args[0])
+    return f"MOD(DAYOFWEEK({arg}) + 5, 7)"
+
+
+def _date_add(translator: ExprTranslator, op: ops.temporal.DateAdd) -> str:
+    date, interval = op.left, op.right
+    date_translated = translator.translate(date)
+    interval_translated = translator.translate(interval)
+    return f"{date_translated} + {interval_translated}"
+
+
+def _date_diff(translator: ExprTranslator, op: ops.temporal.DateDiff) -> str:
+    raise com.UnsupportedOperationError("DATE_DIFF is not supported in Flink.")
+
+
+def _date_from_ymd(translator: ExprTranslator, op: ops.temporal.DateFromYMD) -> str:
+    year, month, day = op.year, op.month, op.day
+    date_string = f"{year.value}-{month.value}-{day.value}"
+    return f"CAST('{date_string}' AS DATE)"
+
+
+def _date_sub(translator: ExprTranslator, op: ops.temporal.DateSub) -> str:
+    date, interval = op.left, op.right
+    date_translated = translator.translate(date)
+    interval_translated = translator.translate(interval)
+    return f"{date_translated} - {interval_translated}"
+
+
+def extract_epoch_seconds(translator: ExprTranslator, op: ops.Node) -> str:
+    arg = translator.translate(op.arg)
+    return f"UNIX_TIMESTAMP(CAST({arg} AS STRING))"
```

Comment:
Nice; this translation makes sense (think there was something non-working before had removed it in between).